Automattic/wp-calypso
# Gutenberg: Add a8c-blocks

Pull in blocks from the `a8c-blocks` repo -- the idea is to eventually get rid of that separate repo and build its blogs from within Calypso instead.

Except for the Idioticon block, which will require a separat audit.

We're also renaming the `p2` preset to `o2`. The reason is that we're using the preset name to generate our style namespacing, and since P2s (well, the O2 plugin that they use) already add the `o2` class to `<body />`, we can piggy-back on that.

TODO of sorts: Bundle size is ~96k for `o2-editor.js`, i.e. roughly 3 times as much as the previous, `create-gutenberg-block` generated  `a8c-blocks` bundles.
Prepending `EMIT_STATS=true` to the build command, followed by running `npx webpack-bundle-size-analyzer stats.json`, reveals that the bundle pulls in `core-js`. This is confirmed by disabling `@babel/transform-runtime` in `babel.config.js`, which produces a build that is only ~48k in size. 

To test:
1. `npm run sdk:gutenberg -- --editor-script=client/gutenberg/extensions/presets/o2/editor --view-script=client/gutenberg/extensions/presets/o2/view -o client/gutenberg/extensions/presets/o2/dist/`
2. Log into your dotcom sandbox, determine its IP, point `testp2a8c dot wordpress dot com` to it (in your `/etc/hosts`)
3. Apply D16966-code
4. Copy the contents of `client/gutenberg/extensions/presets/o2/dist` into your sandbox' `~/public_html/wp-content/a8c-plugins/a8c-blocks/dist`.
5. In your browser, load `testp2a8c dot wordpress dot com`. Make sure it shows up as sandboxed, i.e. there's this badge in the bottom left corner:
![image](https://user-images.githubusercontent.com/96308/43772330-0b755de2-9a42-11e8-8413-fdbd341d8788.png)
6. In the p2tenberg thing on top of the P2, try inserting each of the a8c blocks:
    * Editor notes
    * Prev/next links
    * TODO
    * Trials

Verify that they all work, and are styled properly.
